### PR TITLE
React navigation header

### PR DESCRIPTION
**Pull request by lucasnestrojil**
## Summary
- Added common header background
- Added common back image button

QA Note: The position of title is related to the button in Android, so ever is aling to left.

## Android
header

Galaxy Nexus (720x1280)
image

Pixel 2 (1080x1290)
image

Pixel 3 (1080x2160)
image

Pixel 3 XL (1440x2960)
image

## Trello Card
https://trello.com/c/riM9sgps/15-header

## InVision Link
https://projects.invisionapp.com/d/main#/console/10158364/217697991/preview

### CODE REVIEW

File: navigation/AppNavigation.js [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+      screen: BookDetailContainer,
+      navigationOptions: {
+        title: 'Book Detail'
+      }
+    }
+  },
+  {
+    initialRouteName: 'library',
+    defaultNavigationOptions: {
+      headerBackground: <ImageHeader />,
```
Comment (DLotero) [inline navigation/AppNavigation.js]: for the custom header component I think it'd be better to use `headerTitle` instead. Also I don't think you need to use jsx for this. Try `headerTitle: ImageHeader` and see if it works

File: navigation/AppNavigation.js [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+      headerTitleStyle: styles.titleStyle,
+      headerBackImage: <BackButtonHeader />
```
Comment (DLotero) [inline navigation/AppNavigation.js]: same with `headerBackImage` you don't really need the jsx. Check https://github.com/react-navigation/react-navigation/blob/2c7187b22aeff1cdec5ca6aeebb40c9c798c0888/examples/NavigationPlayground/js/StackWithCustomHeaderBackImage.js

File: navigation/components/BackButtonHeader/styles.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export const styles = StyleSheet.create({
+  image: {
+    margin: 0,
+    width: 20,
+    height: 20
+  }
+});
```
Comment (DLotero) [inline navigation/components/BackButtonHeader/styles.js]: I'm not really sure this file is necessary, can you use `headerBackTitleStyle` in `AppNavigation.js` and see if the back button looks the same?

File: navigation/components/ImageHeader/styles.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export const styles = StyleSheet.create({
+  image: {
+    marginTop: -20,
+    width: '100%',
+    height: 110
+  }
+});
```
Comment (DLotero) [inline navigation/components/ImageHeader/styles.js]: same as the previous comment but use `headerTitleStyle`

File: navigation/styles.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from 'react-native';
+
+import { white } from '../scss/colors';
+
+export const styles = StyleSheet.create({
+  titleStyle: {
+    fontWeight: 'bold',
+    color: white
+  }
+});
```
Comment (DLotero) [inline navigation/styles.js]: this file is not necessary either. try using something like:
```
header: {
	    	titleStyle: {
	    		fontFamily: 'test'
	    	}
	    }    
```

File: navigation/AppNavigation.js (hunk 1 above)
Comment (lucasnestrojil) [reply navigation/AppNavigation.js]: i didn't find the way to solve this. `headerBackground` wait a component to render, if i pass a function this prop will fail.

File: navigation/components/BackButtonHeader/styles.js (hunk 3 above)
Comment (lucasnestrojil) [reply navigation/components/BackButtonHeader/styles.js]: This prop doesnt work for me, i need these styles by reduce the image size. :disappointed:

File: navigation/components/ImageHeader/styles.js (hunk 4 above)
Comment (lucasnestrojil) [reply navigation/components/ImageHeader/styles.js]: same, i need it because on larges resolution the header image truncate

**Pull request merged**